### PR TITLE
Fixes #20661 - use separate queue for rex jobs

### PR DESCRIPTION
**Pull request by iNecas**
Requires https://github.com/Dynflow/dynflow/pull/277, https://github.com/theforeman/foreman-tasks/pull/319 and https://github.com/theforeman/foreman/pull/5320

### CODE REVIEW

File: app/models/setting/remote_execution.rb [hunk 1]
```diff
@@ -63,7 +63,11 @@ def self.load_defaults
                  nil,
                  N_('Default SSH key passphrase'),
                  nil,
-                 { :encrypted => true })
+                 { :encrypted => true }),
+        self.set('remote_execution_workers_pool_size',
```
Comment (adamruzicka) [inline app/models/setting/remote_execution.rb]: It should be probably state somwhere that the user has to restart httpd+foreman-tasks services in production deployment in order for the changes to come into effect.

File: app/lib/actions/remote_execution/run_host_job.rb [hunk 2]
```diff
@@ -7,6 +7,10 @@ class RunHostJob < Actions::EntryAction
       middleware.do_not_use Dynflow::Middleware::Common::Transaction
       middleware.use Actions::Middleware::HideSecrets
 
+      def queue
+        :remote_execution
```
Comment (adamruzicka) [inline app/lib/actions/remote_execution/run_host_job.rb]: How about replacing all the occurences of `:remote_execution` with a constant?

File: app/models/setting/remote_execution.rb (hunk 1 above)
Comment (iNecas) [reply app/models/setting/remote_execution.rb]: Good call, added the info into the description

File: app/lib/actions/remote_execution/run_host_job.rb (hunk 2 above)
Comment (iNecas) [reply app/lib/actions/remote_execution/run_host_job.rb]: Good idea, updated.

**Comment by iNecas** [comment]
Updated

**Comment by adamruzicka** [comment]
Could you please bump the Dynflow dependency again?

**Comment by iNecas** [comment]
Updated to `~> 1.0`, as if we're following the semantic versioning, 1.y bumps should still be fine

**Comment by adamruzicka** [comment]
[test]

**Comment by iNecas** [comment]
Running tests manually http://ci.theforeman.org/job/test_plugin_pull_request/4853/

**Comment by iNecas** [comment]
This should be ready for final review

**Comment by adamruzicka** [comment]
Thanks @iNecas !

**Pull request merged**